### ISSUE TITLE
Onnx export imgsz not being used

### ISSUE THREAD

**Issue by HannoJacobs** [ultralytics/ultralytics#4649]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,

When I try to export my YOLOv8 retrained pytorch model that has been retained with Imgsz=416 the onnx model doesn't seem to use the imgsz parameter and defaults to looking for 640 as the imgsz. The best.pt (imgsz=416) model works if it is used as is but when converted to onnx it doesn't work for inferencing: `model.export(format='onnx', imgsz=(416, 416))`

The error I get is:

```
WARNING ⚠️ Unable to automatically guess model task, assuming 'task=detect'. Explicitly define task for your model, i.e. 'task=detect', 'segment', 'classify', or 'pose'.
Loading model_saves/load_onnx_model/onnx_model/model.onnx for ONNX Runtime inference...

Traceback (most recent call last):
  File "/home/azureuser/ultralytics-usage-template/misc/load_models_from_databricks.py", line 98, in <module>
    load_onnx_model(model_save_folder, run_ID)
  File "/home/azureuser/ultralytics-usage-template/misc/load_models_from_databricks.py", line 86, in load_onnx_model
    detections = loaded_model.predict(image)
  File "/home/azureuser/ultralytics-usage-template/.venv/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/azureuser/ultralytics-usage-template/.venv/lib/python3.10/site-packages/ultralytics/yolo/engine/model.py", line 255, in predict
    return self.predictor.predict_cli(source=source) if is_cli else self.predictor(source=source, stream=stream)
  File "/home/azureuser/ultralytics-usage-template/.venv/lib/python3.10/site-packages/ultralytics/yolo/engine/predictor.py", line 188, in __call__
    return list(self.stream_inference(source, model))  # merge list of Result into one
  File "/home/azureuser/ultralytics-usage-template/.venv/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 35, in generator_context
    response = gen.send(None)
  File "/home/azureuser/ultralytics-usage-template/.venv/lib/python3.10/site-packages/ultralytics/yolo/engine/predictor.py", line 244, in stream_inference
    preds = self.model(im, augment=self.args.augment, visualize=visualize)
  File "/home/azureuser/ultralytics-usage-template/.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/azureuser/ultralytics-usage-template/.venv/lib/python3.10/site-packages/ultralytics/nn/autobackend.py", line 323, in forward
    y = self.session.run(self.output_names, {self.session.get_inputs()[0].name: im})
  File "/home/azureuser/ultralytics-usage-template/.venv/lib/python3.10/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 217, in run
    return self._sess.run(output_names, input_feed, run_options)
onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Got invalid dimensions for input: images for the following indices
 index: 2 Got: 640 Expected: 416
 index: 3 Got: 640 Expected: 416
 Please fix either the inputs or the model.
```
 
 I have tried using dynamic=True and that seems to work but then the inferencing appears to be run at 640x640 which is not what I want (I want 416x416). There also seems to be a comment in the onnx export code that states that dynamic=True only works on CPU
 
 ```
##############################
######## Test onnx model #####
image size = 416
dynamic_ = True
##############################


WARNING ⚠️ Unable to automatically guess model task, assuming 'task=detect'. Explicitly define task for your model, i.e. 'task=detect', 'segment', 'classify', or 'pose'.
Loading runs/416_bug_test/416_YOLOv8n_best.onnx for ONNX Runtime inference...

0: 640x640 2 persons, 9 vests, 43.5ms
Speed: 3.2ms preprocess, 43.5ms inference, 0.5ms postprocess per image at shape (1, 3, 640, 640)
predictions = tensor([[0.0000e+00, 3.9497e+02, 3.2815e+02, 1.1097e+03, 5.1030e-01, 0.0000e+00],
        [2.4608e+03, 0.0000e+00, 2.7060e+03, 6.7322e+02, 4.8907e-01, 0.0000e+00],
        [2.4079e+03, 1.1443e+03, 2.7060e+03, 1.8192e+03, 4.7171e-01, 1.0000e+00],
        [2.4055e+03, 1.0532e+03, 2.7060e+03, 1.6530e+03, 3.9470e-01, 1.0000e+00],
        [2.5199e+03, 7.2700e+01, 2.7060e+03, 3.8118e+02, 3.7301e-01, 1.0000e+00],
        [2.4442e+03, 1.2192e+03, 2.7060e+03, 1.7208e+03, 3.6067e-01, 1.0000e+00],
        [2.4408e+03, 1.1446e+03, 2.7060e+03, 1.5988e+03, 3.5549e-01, 1.0000e+00],
        [6.3858e+02, 1.9437e+03, 1.0354e+03, 2.4599e+03, 2.8484e-01, 1.0000e+00],
        [2.1839e+03, 1.0454e+03, 2.6260e+03, 1.7365e+03, 2.7097e-01, 1.0000e+00],
        [2.4417e+03, 1.2638e+03, 2.7060e+03, 1.8634e+03, 2.6720e-01, 1.0000e+00],
        [2.3357e+03, 1.1406e+03, 2.7060e+03, 1.8956e+03, 2.6720e-01, 1.0000e+00]])
```
 
 # The code to reproduce
I am working with python3.10
```
from ultralytics import YOLO
import cv2

def test_onnx_model():
    onnx_path   = f"runs/416_bug_test/416_YOLOv8n_best.onnx"
    loaded_model = YOLO(onnx_path)
    image = cv2.imread("misc/image_2.jpg")
    detections = loaded_model.predict(image)
    predictions = detections[0].boxes.data
    print(f"predictions = {predictions}")

model = YOLO("runs/416_bug_test/416_YOLOv8n_best.pt")

model.export(format='onnx', imgsz=(416, 416), dynamic=True)
test_onnx_model()
"""
This works but the inferencing is done in 640x640
0: 640x640 2 persons, 9 vests, 39.6ms
Speed: 3.5ms preprocess, 39.6ms inference, 0.6ms postprocess per image at shape (1, 3, 640, 640)
predictions = tensor(
[[0.0000e+00, 3.9497e+02, 3.2815e+02, 1.1097e+03, 5.1030e-01, 0.0000e+00],
[2.4608e+03, 0.0000e+00, 2.7060e+03, 6.7322e+02, 4.8907e-01, 0.0000e+00],
[2.4079e+03, 1.1443e+03, 2.7060e+03, 1.8192e+03, 4.7171e-01, 1.0000e+00],
[2.4055e+03, 1.0532e+03, 2.7060e+03, 1.6530e+03, 3.9470e-01, 1.0000e+00],
"""

model.export(format='onnx', imgsz=(416, 416), dynamic=False)
test_onnx_model()
"""
Above line error when inferencing:
index: 2 Got: 640 Expected: 416
index: 3 Got: 640 Expected: 416
Please fix either the inputs or the model.
"""
```

**Comment by HannoJacobs** [comment]
You have solved this issue in: https://github.com/ultralytics/ultralytics/issues/4652

thanks!

**Comment by glenn-jocher** [comment]
@HannoJacobs hello,

We appreciate your interest in the YOLOv8 project. I'm glad to hear that the issue has been successfully addressed as shared in the issue #4652. 

The team continuously works on improving the project and resolving the reported issues. It's great to know that you found the solution useful!

Should you have any other inquiries or face any problems, don't hesitate to reach out or report new issues. We're here to assist!

Thank you for being part of our GitHub community. Your collaboration helps us make the project better!
